apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/Constants.java

```diff
@@ -18,10 +18,118 @@
+            "jakarta.servlet.ServletInputStream",
+            "jakarta.servlet.ServletConfig");
+
+        private final String servletClass;
+        private final String filterClass;
+
+        private final String requestClass;
+        private final String responseClass;
+
+        private final String filterChainClass;
+        private final String servletInputStreamClass;
+        private final String servletConfigClass;
+
+        // async context class has been introduced in servlet spec 3.0
+        // choice of class name to use for this test does not work as expected across all application servers
+        // for example, 'javax.servlet.annotation.WebServlet' annotation is not working as expected on Payara
+        private final String asyncContextClass;
+
+        // OSGi bundles might only include 'jakarta.servlet' and not 'jakarta.servlet.http' (same for `java.servlet`), hence
+        // this allows to disable instrumentation as it is required for our instrumentation code.
+        private final String osgiClassloaderFilterClass;
+
+        ServletImpl(String servletClass, String filterClass, String requestClass, String responseClass, String filterChainClass,
+                    String asyncContextClass, String osgiClassloaderFilterClass, String servletInputStreamClass, String servletConfigClass) {
+            this.servletClass = servletClass;
+            this.filterClass = filterClass;
+            this.requestClass = requestClass;
+            this.responseClass = responseClass;
+            this.filterChainClass = filterChainClass;
+            this.asyncContextClass = asyncContextClass;
+            this.osgiClassloaderFilterClass = osgiClassloaderFilterClass;
+            this.servletInputStreamClass = servletInputStreamClass;
+            this.servletConfigClass = servletConfigClass;
+        }
+
+        public ElementMatcher.Junction<NamedElement> servletClass() {
+            return named(servletClass);
+        }
+
+        public ElementMatcher.Junction<NamedElement> filterClass() {
+            return named(filterClass);
+        }
+
+        public ElementMatcher.Junction<NamedElement> requestClass() {
+            return named(requestClass);
+        }
+
+        public ElementMatcher.Junction<NamedElement> responseClass() {
+            return named(responseClass);
+        }
+
+        public ElementMatcher.Junction<NamedElement> filterChainClass() {
+            return named(filterChainClass);
+        }
+
+        public ElementMatcher.Junction<NamedElement> asyncContextClass() {
+            return named(asyncContextClass);
+        }
+
+        public ElementMatcher.Junction<TypeDescription> servletInputStreamClass() {
+            return named(servletInputStreamClass);
+        }
+
+        public ElementMatcher.Junction<TypeDescription> servletConfigClass() {
+            return named(servletConfigClass);
+        }
+
+        public ElementMatcher.Junction<ClassLoader> getClassloaderFilter() {
```

Comment:
[minor] consider ending these accessor names with `Matcher`, for example `servletConfigClassMatcher()`.